src/openms/source/QC/MzCalibration.cpp

```diff
@@ -0,0 +1,193 @@
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Juliane Schmachtenberg $
+// --------------------------------------------------------------------------
+
+
+#include <OpenMS/QC/QCBase.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/QC/MzCalibration.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CONCEPT/Exception.h>
+#include <OpenMS/METADATA/DataProcessing.h>
+#include <OpenMS/MATH/MISC/MathFunctions.h>
+
+using namespace std;
+ 
+namespace OpenMS
+{
+  MzCalibration::MzCalibration() : mz_raw_{}, mz_ref_{}, no_mzml_(false), name_("MzCalibration")
+  {}
+  // find original m/z Value, set meta value "mz_raw" and set meta value "mz_ref"
+  void MzCalibration::compute(FeatureMap& features, const MSExperiment& exp)
+  {
+    if (features.empty())
+    {
+      LOG_WARN << "The FeatureMap is empty.\n";
+    }
+    if (exp.empty())
+    {
+      no_mzml_ = true;
+      LOG_WARN << "Metric MzCalibration received an empty mzml file. Only reporting uncalibrated mz error.\n";
+    }
+    else
+    {
+      no_mzml_ = false;
+      // check for Calibration
+      auto is_not_elem = [](const boost::shared_ptr<const OpenMS::DataProcessing> &dp)
+      {
+        return (dp->getProcessingActions().count(DataProcessing::CALIBRATION) == 0);
+      };
+      if (all_of(exp[0].getDataProcessing().begin(), exp[0].getDataProcessing().end(), is_not_elem))
+      {
+        no_mzml_ = true;
+        LOG_WARN << "Metric MzCalibration received an mzml file which did not undergo InternalCalibration. Only reporting uncalibrated mz error.\n";
+      }
+    }
+
+    // set meta values for the first hit of all PeptideIdentifications of all features
+    for (Feature& feature : features)
+    {
+      if (feature.getPeptideIdentifications().empty())
+      {
+        LOG_WARN << "A Feature is empty.\n";
```

Comment:
remove the warning...

Reply:
if fact, you can remove the whole if-block. The iteration below with just do nothing.. which is fine